lib/matplotlib/tests/test_contour.py

```diff
@@ -715,3 +715,12 @@ def test_bool_autolevel():
     assert plt.tricontour(x, y, z).levels.tolist() == [.5]
     assert plt.tricontourf(x, y, z.tolist()).levels.tolist() == [0, .5, 1]
     assert plt.tricontourf(x, y, z).levels.tolist() == [0, .5, 1]
+
+
+def test_contour_all_nan():
+    # Smoke test for gh#14124
+    x = np.array([[np.nan, np.nan], [np.nan, np.nan]])
+    with pytest.warns(UserWarning) as w:
```

Comment:
You should add the match string here rather than asserting on private message lists. https://docs.pytest.org/en/7.1.x/how-to/capture-warnings.html#warns

Reply:
There are two warnings, so this should be the way to go. 
https://docs.pytest.org/en/7.1.x/how-to/capture-warnings.html#recwarn

Reply:
As @oscargus mentioned, There are two warnings, so I manually checked the warning messages for both.

Yes, the warning message is coming from numpy.
```bash
/usr/lib/python3/dist-packages/numpy/ma > grep "Warning: converting a masked" core.py   
            warnings.warn("Warning: converting a masked element to nan.", stacklevel=2)
```

Reply:
The warning definitely comes from numpy. It is converting the floats for zmin/zmax `float(np.ma.masked)`

Ahh, now I see that (maybe unfortunately?) the levels actually have values now, so we only have the two warnings (there used to be other level related warnings on earlier versions):
`[-1.0e-13 -7.5e-14 -5.0e-14 -2.5e-14  0.0e+00  2.5e-14  5.0e-14  7.5e-14 1.0e-13]`

I guess these warnings don't really help much for the root of the problem, so I'd actually suggest getting rid of them altogether by converting the typecasts to: `z.min().astype(float)`, and think about instead giving a more helpful error message somewhere else.

Reply:
So, should we close this PR and add `z.min().astype(float)` & `z.max().astype(float)` to a new one?
I'm getting the correct result using `astype`

Reply:
That seems like a reasonable fix to me and seems to line up with the `plt.plot([np.nan], [np.nan])` case not issuing any warnings either.